lib/random.gd

```diff
@@ -237,6 +259,70 @@ fi;
 DeclareOperation( "RandomSource", [IsOperation] );
 DeclareOperation( "RandomSource", [IsOperation, IsObject] );
 
+
+##############################################################################
+##
+##  <#GAPDoc Label="RandomSource_develop">
+##  <Subsection Label="Implementing new kinds of random sources">
+##  <Heading>Implementing new kinds of random sources</Heading>
+##
+##  If one wants to implement a new kind of random sources then
+##  the first step is the declaration of a new category <C>C</C>, say,
+##  that implies <Ref Filt="IsRandomSource"/>, analogous to the categories
+##  listed in Section <Ref Subsect="Kinds of Random Sources"/>.
+##  <P/>
+##  Then the following method installations are needed.
+##  <P/>
+##  <Index Key="Init"><C>Init</C> (initialize a random source object)</Index>
+##  <C>InstallMethod( Init, [ C, IsObject ], function( prers, seed_or_state )
+##  ... end );</C>
+##  <P/>
+##  Here <C>prers</C> is an empty component object (which has already the
+##  filter <C>C</C>), and <C>seed_or_state</C> is an integer or a state value
+##  as returned by <Ref Oper="State"/> that describes the initial state of
+##  the random source.
+##  The function should fill in the actual data and then return the
+##  (now initialized) object <C>prers</C>.
+##  The default used for <C>seed_or_state</C> is the integer <C>1</C>.
+##  A given state value need not be copied by the function.
+##  <P/>
+##  <C>InstallMethod( Random, [ C, IsInt, IsInt ], function( rs, low, high )
+##  ... end );</C>
+##  <P/>
+##  Here <C>rs</C> is an already initialized random source object in the
+##  category <C>C</C>, and the function returns an integer between <C>low</C>
+##  and <C>high</C> (inclusive).
+##  It is not defined what happens when <C>low</C> is larger than <C>high</C>.
+##  <P/>
+##  <C>InstallMethod( State, [ C ], function( rs ) ... end );</C>
+##  <P/>
+##  If <C>rs</C> supports resetting then the function must return an object
+##  that describes the current state of <C>rs</C>.
+##  This object must be an independent copy, that is,
+##  calling <Ref Oper="Random" Label="for random source and list"/>
+##  for <C>rs</C> must not change the object that was returned by
+##  <Ref Oper="State"/>;
+##  otherwise <Ref Func="ReturnFail"/> should be installed.
+##  <P/>
+##  <C>InstallMethod( Reset, [ C, IsObject ], function( rs, seed_or_state )
+##  ... end );</C>
+##  <P/>
+##  If <C>rs</C> supports resetting then the function must reinitialize
+##  <C>rs</C> to the integer or <Ref Oper="State"/> value <C>seed_or_state</C>
+##  and must return the <Ref Oper="State"/> value of <C>rs</C> before these
+##  changes; if resetting is not supported then <Ref Func="ReturnNothing"/>
+##  should be installed.
+##  Reset need not copy a given state.
+##  Note that the generic unary <Ref Oper="Reset"/> method uses the default
+##  seed <C>1</C>.
+##  <P/>
+##  An example of an implementation as described here is given by the
```

Comment:
of course the library implementations for `IsGAPRandomSource` and `IsMersenneTwister` are also examples... :-)

Reply:
The method installations for `IsMersenneTwister` fit.
Concerning `IsGAPRandomSource`, the code does not follow the recommended pattern: A `Random` method for a list gets installed, which relies on a default method that delegates the two-integer variant to the list variant.